lib/extensions.js

```diff
@@ -76,21 +76,40 @@ function getBinaryName() {
 }
 
 /**
- * Retrieve the URL to fetch binary.
- * If environment variable SASS_BINARY_URL
- * is set, return that path. Otherwise make
- * path using current release version and
- * binary name.
+ * Determine the URL to fetch binary file from.
+ * By default feth from the node-sass distribution
+ * site on GitHub.
+ * 
+ * The default URL can be overriden using
+ * the environment variable SASS_BINARY_SITE
+ * or a command line option --sass-binary-site:
+ *
+ *   node scripts/install.js --sass-binary-site http://example.com/
+ *
+ * The URL should to the mirror of the repository
+ * laid out as follows:
+ *
+ * SASS_BINARY_SITE/
+ * 
+ *  v3.0.0-beta.4
+ *  v3.0.0-beta.4/freebsd-x64-14_binding.node
+ *  ....
+ *  v3.0.0-beta.5
+ *  v3.0.0-beta.5/freebsd-ia32-11_binding.node
+ *  v3.0.0-beta.5/freebsd-x64-42_binding.node
+ *  ... etc. for all supported versions and platforms
  *
  * @api private
  */
 
 function getBinaryUrl() {
-  return flags['--sass-binary-url'] ||
-         package.nodeSassConfig ? package.nodeSassConfig.binaryUrl : null ||
-         process.env.SASS_BINARY_URL ||
-         ['https://github.com/sass/node-sass/releases/download/v',
-          package.version, '/', sass.binaryName].join('');
+  var site = flags['--sass-binary-site'] || process.env.SASS_BINARY_SITE  ||
+             (package.nodeSassConfig ? package.nodeSassConfig.binarySite : null);
+  if (site) {
+		  return [site, 'v' + package.version, sass.binaryName].join('/');
+  } else {
+          return null;
```

Comment:
This shoyldn't be possible. We should default to `'https://github.com/sass/node-sass/releases/download/v` if the users has give no proeference.